config/confighttp/compression.go

```diff
@@ -117,6 +118,25 @@ func httpContentDecompressor(h http.Handler, eh func(w http.ResponseWriter, r *h
 				}
 				return zr, nil
 			},
+			"snappy": func(body io.ReadCloser) (io.ReadCloser, error) {
+
+				compressed, err := io.ReadAll(body)
+				if err != nil {
+					return nil, err
+				}
+
+				reqBuf, err := snappy.Decode(nil, compressed)
+				if err != nil {
+					return nil, err
+				}
+
+				sr := &readCloser{
+					reader: bytes.NewReader(reqBuf),
+					closer: body,
+				}
```

Comment:
You already read the bytes why not closing the body and make the closer no-op?

Reply:
I address this.
Nice suggestion, no need to create my own struct either.